src/types.js

```diff
@@ -53,6 +53,24 @@ export function toArray(arrayLike) {
 }
 
 /**
+ * Returns a map-like object.
+ * If opt_initial is omitted returns an objects with a null prototype.
+ * Otherwise takes the given object and sets its prototype to null.
+ * @param {T=} opt_initial This should typically be an object literal.
+ * @return {T}
+ * @template T
+ */
+export function map(opt_initial) {
```

Comment:
Is this really preferable to polyfilling the ES6 `Map`?  It looks like that's already supported in Chrome, Safari, Firefox, and mostly in IE 11.